From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 6 Jun 2020 21:07:15 -0500
Subject: [PATCH] bound villager trade maxUses by 16

Older versions of Minecraft expanded the maxUses when unlocking trades
Modern version resets the number of uses

diff --git a/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java b/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java
index 8a9a701baabdaf066cd9b28c05430f673fcafb4e..b656842b002462dd33b1b2b6b66d244be06c4151 100644
--- a/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java
+++ b/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java
@@ -53,7 +53,7 @@ public class MerchantOffer {
         this.result = ItemStack.of(nbt.getCompound("sell"));
         this.uses = nbt.getInt("uses");
         if (nbt.contains("maxUses", 99)) {
-            this.maxUses = nbt.getInt("maxUses");
+            this.maxUses = Math.min(16, nbt.getInt("maxUses")); // EMC
         } else {
             this.maxUses = 4;
         }
